lib/Service/RoomService.php

```diff
@@ -61,6 +66,10 @@ public function createOneToOneConversation(IUser $actor, IUser $targetUser): Roo
 			throw new InvalidArgumentException('invalid_invitee');
 		}
 
+		if (!$this->shareManager->currentUserCanEnumerateTargetUser($actor, $targetUser)) {
+			throw new RoomNotFoundException();
+		};
```

Comment:
should be inside the catch below, so that existing conversations can still be reopened